src/nupyprop/geometry_v2.py

```diff
@@ -0,0 +1,444 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed May 26 11:55:30 2021
+
+@author: sam
+    
+Updated Wed April 10 14:02:27 2024
+    
+@author: luke
+"""
+
+import nupyprop.data as Data
+#from nupyprop.propagate import geometry as Geometry
+import nupyprop.constants as const
+
+from collections import OrderedDict
+import numpy as np
+from scipy import integrate
+import sympy as sp
+from astropy.table import Table
+#import multiprocessing as mp
+#from multiprocessing import Pool
+import warnings
+warnings.filterwarnings('ignore')
+
+#Earth Radius in km
+Re = const.R_earth
+
+#PREM layers based on R_Earth
+Rlay = const.Rlay
+
+#Water density in g/cm^3
+rho_water = const.rho_water
+
+#Earth Emergence angles in steps of 0.1 degrees
+beta_arr = const.beta_arr
+
+#adding from geometry fortran module
+
+def premdensity(Rin , idepth):
+    """
+    Calculates the density at a radius inside the Earth according to the PREM model.
+    hep-ph/9512364v1 eq. 25
+        
+    Args:
+        Rin (float):
+            Radius (in km) at which density is to be calculated.
+        idepth (integer):
+            Depth (in km) of water layer (sets the last layer).
+
+    Returns:
+        float:
+            Density (in g/cm^3) at the specified radius.
+    """
+
+    #sets the last layer as water layer
+    Rlay[9] = 6368.0 + (3.0 - float(idepth))
+    
+    y = Rin/Re
+    
+    if Rin <= Rlay[0]:
+        edens = 13.0885-8.8381*y**2
+    elif Rin <= Rlay[1]:
+        edens = 12.5815-1.2638*y-3.6426*y**2-5.5281*y**3
+    elif Rin <= Rlay[2]:
+        edens = 7.9565-6.4761*y+5.5283*y**2-3.0807*y**3
+    elif Rin <= Rlay[3]:
+        edens = edens = 5.3197-1.4836*y
+    elif Rin <= Rlay[4]:
+        edens = 11.2494-8.0298*y
+    elif Rin <= Rlay[5]:
+        edens = 7.1089-3.8045*y
+    elif Rin <= Rlay[6]:
+        edens = 2.6910+0.6924*y
+    elif Rin <= Rlay[7]:
+        edens = 2.900
+    elif Rin <= Rlay[8]:
+        edens = 2.600
+    elif Rin <= Rlay[9]:
+        edens = 1.020
+    elif Rin <=Rlay[9]*1.001:
+        edens = 1.020
+    else:
+        edens=0.0
```

Comment:
This looks very verbose, you can maybe define all these conditions in a tuple and then use a loop to match for the corresponding Rin. Hint: Try using ChatGPT :)